AppCenterDataStorage/AppCenterDataStorage/MSPaginatedDocuments.m

```diff
@@ -2,25 +2,74 @@
 // Licensed under the MIT License.
 
 #import "MSPaginatedDocuments.h"
+#import "MSCosmosDb.h"
+#import "MSDataStore.h"
+#import "MSDataStoreInternal.h"
+#import "MSDataStorePrivate.h"
+#import "MSPaginatedDocumentsInternal.h"
 #import "MSSerializableDocument.h"
+#import "MSTokenExchange.h"
 
 @implementation MSPaginatedDocuments
 
-- (BOOL)hasNextPage {
-  // @todo
-  return NO;
+@synthesize currentPage = _currentPage;
+@synthesize continuationToken = _continuationToken;
+@synthesize partition = _partition;
+@synthesize documentType = _documentType;
+@synthesize readOptions = _readOptions;
+
+- (instancetype)initWithPage:(MSPage *)page
+                   partition:(NSString *)partition
+                documentType:(Class)documentType
+                 readOptions:(MSReadOptions *_Nullable)readOptions
+           continuationToken:(NSString *_Nullable)continuationToken {
+  if ((self = [super init])) {
+    _currentPage = page;
+    _partition = partition;
+    _documentType = documentType;
+    _readOptions = readOptions;
+    _continuationToken = continuationToken;
+  }
+  return self;
+}
+
+- (instancetype)initWithPage:(MSPage *)page {
+  if ((self = [super init])) {
+    _currentPage = page;
+    _continuationToken = nil;
+  }
+  return self;
 }
 
-- (MSPage<id<MSSerializableDocument>> *)currentPage {
-  // @todo
-  return nil;
+- (instancetype)initWithError:(MSDataSourceError *)error {
+  if ((self = [super init])) {
```

Comment:
Here you would `MSPage *pageWithError = [[MSPage alloc] initWithError:error];` before the `if` and call:
```suggestion
  if ((self = [self initWithPage:pageWithError])) {
```